FirebaseAppCheck/Apps/FIRAppCheckTestApp/FIRAppCheckTestApp/AppDelegate.swift

```diff
@@ -26,11 +26,45 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
                      .LaunchOptionsKey: Any]?) -> Bool {
     FirebaseApp.configure()
 
+    requestDeviceCheckToken()
+
+    requestDebugToken()
+
+    if #available(iOS 14.0, *) {
+      requestAppAttestToken()
+    }
+
+    return true
+  }
+
+  // MARK: UISceneSession Lifecycle
+
+  func application(_ application: UIApplication,
+                   configurationForConnecting connectingSceneSession: UISceneSession,
+                   options: UIScene.ConnectionOptions) -> UISceneConfiguration {
+    // Called when a new scene session is being created.
+    // Use this method to select a configuration to create the new scene with.
+    return UISceneConfiguration(
+      name: "Default Configuration",
+      sessionRole: connectingSceneSession.role
+    )
+  }
+
+  func application(_ application: UIApplication,
+                   didDiscardSceneSessions sceneSessions: Set<UISceneSession>) {
+    // Called when the user discards a scene session.
+    // If any sessions were discarded while the application was not running, this will be called shortly after application:didFinishLaunchingWithOptions.
+    // Use this method to release any resources that were specific to the discarded scenes, as they will not return.
+  }
+
+  // MARK: App Check providers
+
+  func requestDeviceCheckToken() {
     guard let firebaseApp = FirebaseApp.app() else {
-      return true
+      return
     }
 
-    FIRDeviceCheckProvider(app: firebaseApp)?.getToken(completion: { token, error in
+    DeviceCheckProvider(app: firebaseApp)?.getToken(completion: { token, error in
```

Comment:
side note... idea: WDYT about a Swift API extension to use result type here?

Reply:
Good point, I'll update it for consistency. 

Note, that this particular API is not intended to be used by application developers. The method is expected to be called by the App Check core only or for testing purposes (as e.g. here). The API is public mostly because we would like to provide possibility to plug in any provider - one of predefined we have or a custom provider implemented by the app developers. Public API to get Firebase App Check token is a separate task in the backlog. We may consider using Result or/and Combine extension for it but it's out of scope for the current project.

Reply:
> Note, that this particular API is not intended to be used by application developers

I see, I didn't realize. Makes sense, thanks for clarifying!

Reply:
Would we like to add a scheme to the test app that uses the SwiftUI lifecycle? If so, I could help with that in a subsequent PR

Reply:
Good question. I'm not sure if App Check SDK behaves differently with app delegate and SwiftUI lifecycle because it doesn't depend on any lifecycle events yet. But feel free to add any test code you see relevant. Thank you for help!